common.py

```diff
@@ -0,0 +1,110 @@
+            if not local_path.endswith('/gender_novels'):
+                is_local = False
+                print(f"WARNING: The FileLoaderMixin should be placed in the main path of the gender_novels project."
+                      f"It's currently in {local_path}. Until the Mixin is in the correct path, files are loaded " \
+                      f"remotely.")
+        except NameError:
+            is_local = False
+
+        if is_local:
+            print(f'loading {file_path} locally.')
+            return self._load_file_locally(file_path)
+        else:
+            print(f'loading {file_path} remotely')
+            return self._load_file_online(file_path)
+
+
+    def _load_file_locally(self, file_path):
+
+        # I need a way of getting the local path to the base of the repo. This file is currently in the base of the
+        # repo so it returns the correct path. But it will change once this function gets moved.
+        local_base_path = os.path.abspath(os.path.dirname(__file__))
+        file = open(f'{local_base_path}/{file_path}', mode='r')
+
+        if file_path.endswith('.csv'):
+            result = file.readlines()
+        elif file_path.endswith('.txt'):
+            result = file.read()
+
+        file.close()
+        return result
+
+    def _load_file_online(self, file_path):
+
+        base_path = 'https://raw.githubusercontent.com/dhmit/gender_novels/master/'
+        url = f'{base_path}/{file_path}'
+        response = urllib.request.urlopen(url)
+
+        if url.endswith('.csv'):
+            return [line.decode('utf8') for line in response.readlines()]
+        elif url.endswith('.txt'):
+            return response.read().decode('utf8')
+
+class Corpus(FileLoaderMixin):
+
+    def __init__(self, corpus_name):
+        self.corpus_name = corpus_name
+        print("here", type(self))
+        self.novels = self._load_novels()
+
+    def _load_novels(self):
+
+        novels = []
+
+        csv_path = f'corpora/{self.corpus_name}/{self.corpus_name}.csv'
+        csv_file = self.load_file(csv_path)
+        csv_reader = csv.DictReader(csv_file)
+
+        for novel_metadata in csv_reader:
+            novel_metadata['corpus_name'] = self.corpus_name
+            novels.append(Novel(novel_metadata_dict=novel_metadata))
+
+        return novels
+
+class Novel(FileLoaderMixin):
+
+    def __init__(self, novel_metadata_dict):
+        for k, v in novel_metadata_dict.items():
+            setattr(self, k, v)
+
+        # Check that the essential attributes for the novel exists.
+        # Currently available attributes that are not checked are: country_publication, author_gender, and notes.
+        assert hasattr(self, 'author')
+        assert hasattr(self, 'date')
+        assert hasattr(self, 'title')
+        assert hasattr(self, 'corpus_name')
+        assert hasattr(self, 'filename')
+
+
+        if not hasattr(self, 'text'):
+            file_path = f'corpora/{self.corpus_name}/texts/{self.filename}'
```

Comment:
I like using f-strings!  But this won't run properly on all windows systems.  Better:

```
# import pathlib at top
file_path = pathlib.Path('corpora') / self.corpus_name / 'texts' / self.filename
```

or use os.sep.join(['corpora', self.corpus_name, 'texts', self.filename])

Reply:
Done. Thoughts on using forward slashes vs. Path('corpora', self.corpus_name, 'texts', self.filename) ?

Reply:
no difference.  yours is probably better.  I usually don't have all the parts of the Path ready at the moment I make the Path object.

If you need to support Py3.4, then pathlib isn't worth the headaches.  If you're on 3.6+, it's amazing.